### ISSUE TITLE
Shutdown supervisor without supervisorId

### ISSUE THREAD

**Issue by nateMJ** [druid-io/druid#5780]
I am unable to shutdown a supervisor that was created without an ID or topic. If the only way to shutdown supervisors or tasks is using IDs, shouldn't there be an error when a supervisor is posted without an ID or kafka topic?

**Comment by jihoonson** [comment]
Hi @nateMJ, every supervisor should have its ID which is the name of dataSource where the data is ingested to.

**Comment by nateMJ** [comment]
@jihoonson I understand that but posting a supervisor without a datasource name or kafka topic name posts successfully but can't be deleted later. Is that a bug?

This is the schema I posted:
```
{
  "type": "kafka",
  "dataSchema": {
    "dataSource": "${datasource}",
    "parser": {
      "type": "string",
      "parseSpec": {
        "format": "json",
        "timestampSpec": {
          "column": "date",
          "format": "auto"
        },
        "dimensionsSpec": {
          "dimensions": [
            "id",
            "country",
            "region",
            "district"
          ]
    },
    "metricsSpec": [
      {
        "type": "count",
        "name": "count"
      }
    ],
    "granularitySpec": {
      "type": "uniform",
      "segmentGranularity": "day",
      "queryGranularity": "none",
      "rollup": true
    }
  },
  "ioConfig": {
    "topic": "${topic}",
    "taskCount": 15,
    "useEarliestOffset": true,
    "consumerProperties": {
      "bootstrap.servers": "localhost:9092"
    },
    "taskDuration": "PT5M"
  },
  "tuningConfig": {
    "type": "kafka",
    "maxRowsPerSegment": 5000000,
    "resetOffsetAutomatically": true
  }
}

```

**Comment by jihoonson** [comment]
> posting a supervisor without a datasource name or kafka topic name posts successfully but can't be deleted later.

What do you mean by `posting a supervisor without a datasource name`? Does it mean `${datasource}` was empty when you posted the above supervisor spec?

You can't submit a supervisor spec without dataSource name. Druid internally checks it and returns an error if it's empty.

**Comment by Dylan1312** [comment]
Confirmed you can post a supervisor spec with a dataSource as an empty string, at least in Druid 0.12.0.

To remove the supervisor I updated the specId in the metadata store (in the druid_supervisors table), restarted the overlord and then shut it down using the updated specId.

**Comment by jihoonson** [comment]
@Dylan1312 thank you for reporting! This is definitely a bug.

**Comment by jihoonson** [comment]
@Dylan1312 would you make a patch for this bug? I think we should check `dataSource` is null or empty in the constructor of `DataSchema`.

**Comment by jihoonson** [comment]
@Dylan1312 oh, you already raised a PR! I'll take a look. Thanks!

**Comment by jihoonson** [comment]
Fixed in #5785.
